### ISSUE TITLE
Support JSON-formatted secrets in SecretSourceResolver

### ISSUE THREAD

**Issue by chriskilding** [jenkinsci/configuration-as-code-plugin#2017]
### What feature do you want to see added?

I've been thinking on how best to support JSON-formatted secrets (e.g. from AWS Secrets Manager) for a while as you know.

Initially I thought about doing something provider-specific for this. But today I realised that actually the right place for this might be as an additional `json:` prefix handler (and corresponding `JsonLookup.INSTANCE`) in the SecretSourceResolver, alongside the existing ones like `base64:`.

This would allow SecretSourceResolver to handle JSON secrets from **any** SecretSource implementation. It also allows the SecretSource implementations to stay simple and do one thing well - fetching secrets.

It would look like this:

```yaml
${json:<key>:<secret-id>}
```

So for example, if you have a JSON-formatted secret called `my-secret`, and you want to extract the `username` and `password` fields respectively, you'd do this:

```yaml
jenkins:
  securityRealm:
    local:
      allowsSignup: false
      users:
      - id: "${json:username:my-secret}"
        password: "${json:password:my-secret}"
```

And the SecretSource implementation would simply be told to fetch `my-secret`, like it does today.

### Upstream changes

None that I know of

**Comment by chriskilding** [comment]
I was mainly considering the ARN format since that would have given the JSON key a structured place to live. But this feature proposal is a better way of doing it. That means users won't need to look up secrets by ARN just to parse a JSON secret.

The only reason they'd use an ARN with this feature in place would be to fetch a secret from another AWS account. With the bug fix in #2007 the `arn:` prefix should be passed to the SecretSource implementation intact, and the ARN based lookup should therefore work.

**Comment by chriskilding** [comment]
Closing, since we implemented support in #2018
